kornia/augmentation/_2d/intensity/motion_blur.py

```diff
@@ -81,17 +81,23 @@ def __init__(
         self._param_generator = rg.MotionBlurGenerator(kernel_size, angle, direction)
         self.flags = dict(border_type=BorderType.get(border_type), resample=Resample.get(resample))
 
-    def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, Tensor]:
+    def generate_parameters(self, batch_shape: torch.Size):
         params = super().generate_parameters(batch_shape)
-        params["idx"] = torch.randint(batch_shape[0], (1,)).item()
+        params["idx"] = tensor([0]) if batch_shape[0] == 0 else torch.randint(batch_shape[0], (1,))
         return params
 
     def apply_transform(
         self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
     ) -> Tensor:
         # sample a kernel size
         kernel_size_list: List[int] = params["ksize_factor"].tolist()
-        idx: int = cast(int, params["idx"])
+
+        # 1. We have to apply the same kernel size to all samples in the batch, thus we take the previously
+        # selected random index --- `params["idx"][0]` --- to determine the applied kernel size.
+        # 2. The `VideoSequential` flattens the first two dimensions, effectively creating a larger batch.
+        # Its method `VideoSequential.__repeat_param_across_channels__` repeats the previously selected index,
+        # creating a tensor with equal values. Hence, taking the first one (`params["idx"][0]`) is legit.
+        idx: int = cast(int, params["idx"][0])
```

Comment:
`int(params["idx"][0])` doesn't work here?

Reply:
will do the same... I kept it as close as possible to the previous code

Reply:
cc @shijianjian

Reply:
It is a bit different here. ```int(...)``` performs the exact behaviour to convert a tensor to an integer type, while ```cast(...)``` only means to cheat the mypy check. So that in this context, ```cast``` may not be pretty appropriate here.

Reply:
I was not familiar with the `cast(...)`. anyway it should be easy to PR an improvement for this.

Reply:
It should be fine here since the changes you made will not break the PyTorch computation graph. No worries.